manim/typing.py

```diff
@@ -1,133 +1,462 @@
-""" `shape: (N,3,3)` An Array of Quadratic Bezier Handles `[[[float, float, float], [float, float, float], [float, float, float]], ...]`. """
 
-CubicBezierPoints: TypeAlias = npt.NDArray[PointDType]
-""" `shape: (4,3)` An Array of Cubic Bezier Handles `[[float, float, float], [float, float, float], [float, float, float], [float, float, float]]`. """
+"""
+[CATEGORY]
+Vector types
+"""
 
-BezierPoints: TypeAlias = npt.NDArray[PointDType]
-""" `shape: (N,3)` An Array of Cubic Bezier Handles `[[float, float, float], ...]`.
-`N` Is always multiples of the degree of the Bezier curve.
-(Please refer to the documentation of the function you are using for further type Information)
+Vector2D: TypeAlias = Point2D
+"""``shape: (2,)``
+A 2D vector: ``[float, float]``.
+Normally a function or method which expects a `Vector2D` as a parameter can handle being
+passed a `Vector3D` instead.
 """
 
-FlatBezierPoints: TypeAlias = npt.NDArray[PointDType]
-""" `shape: (N)` An Array of Bezier Handles but flattened `[float, ...]`."""
+Vector2D_Array: TypeAlias = Point2D_Array
+"""``shape: (M, 2)``
+An array of 2D vectors: ``[[float, float], ...]``.
+Normally a function or method which expects a `Vector2D_Array` as a parameter can handle being
+passed a `Vector3D_Array` instead.
+"""
 
-Point2D_Array: TypeAlias = npt.NDArray[PointDType]
-""" `shape: (N,2)` An Array of Points in 2D Space `[[float, float], ...]`.
+Vector3D: TypeAlias = Point3D
+"""``shape: (3,)``
+A 3D vector: ``[float, float, float]``.
+"""
 
-(Please refer to the documentation of the function you are using for further type Information)
+Vector3D_Array: TypeAlias = Point3D_Array
+"""``shape: (M, 3)``
+An array of 3D vectors: ``[[float, float, float], ...]``.
 """
 
-InternalPoint3D_Array: TypeAlias = npt.NDArray[PointDType]
-""" `shape: (N,3)` An Array of Points in 3D Space `[[float, float, float], ...]`.
-This type alias is mostly made available for internal use and only includes the numpy type.
+VectorND: TypeAlias = Union[npt.NDArray[PointDType], tuple[float, ...]]
+"""``shape (N,)``
+An `N`-D vector: ``[float, ...]``.
 """
 
-Point3D_Array: TypeAlias = Union[
-    InternalPoint3D_Array, Tuple[Tuple[float, float, float], ...]
+VectorND_Array: TypeAlias = tuple[VectorND, ...]
+"""``shape (M, N)``
+An array of `N`-D vectors: ``[[float, ...], ...]``.
+"""
+
+RowVector: TypeAlias = Union[npt.NDArray[PointDType], tuple[tuple[float, ...]]]
+"""``shape: (1, N)``
+A row vector: ``[[float, ...]]``.
+"""
+
+ColVector: TypeAlias = Union[npt.NDArray[PointDType], tuple[tuple[float], ...]]
+"""``shape: (N, 1)``
+A column vector: ``[[float], [float], ...]``.
+"""
+
+
+"""
+[CATEGORY]
+Matrix types
+"""
+
+MatrixMN: TypeAlias = Union[npt.NDArray[PointDType], tuple[tuple[float, ...], ...]]
+"""``shape: (M, N)``
+A matrix: ``[[float, ...], [float, ...], ...]``.
+"""
+
+Zeros: TypeAlias = Union[npt.NDArray[PointDType], tuple[tuple[Literal[0], ...], ...]]
```

Comment:
IIRC it was added in the original typing PR and is not a modification in this PR.

Reply:
^ this

Reply:
Regardless of when it was introduced, I'll probably wonder whether this really is a useful type alias whenever this file is changed next. I've not been working with it, so I can't tell whether it really is useful -- if it is, no action required. If it is not, a follow-up PR might be a good idea. In the scope of this PR there is nothing to do here.

Reply:
Currently the only two uses of `Zeros` are at `VMobject.get_fill_rgbas()` and `VMobject.get_stroke_rgbas()`. The necessity of this alias is debatable:
```py
    def get_fill_rgbas(self) -> RGBA_Array_Float | Zeros:
        try:
            return self.fill_rgbas
        except AttributeError:
            return np.zeros((1, 4))
```
but, as you mention, if this alias isn't that useful, another PR to handle this would be better :+1: